engines/freescape/gfx.h

```diff
@@ -0,0 +1,173 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef FREESCAPE_GFX_H
+#define FREESCAPE_GFX_H
+
+#include "common/rect.h"
+#include "common/system.h"
+
+#include "graphics/tinygl/pixelbuffer.h"
+#include "math/frustum.h"
+#include "math/matrix4.h"
+#include "math/vector3d.h"
+
+namespace Freescape {
+
+typedef Common::Array<byte *> ColorMap;
+
+class Renderer;
+
+class Texture {
+public:
+	Texture(){};
+	virtual ~Texture(){};
+
+	uint width;
+	uint height;
+	Graphics::PixelFormat format;
+
+	virtual void update(const Graphics::Surface *surface) = 0;
+	virtual void updatePartial(const Graphics::Surface *surface, const Common::Rect &rect) = 0;
+
+	static const Graphics::PixelFormat getRGBAPixelFormat();
```

Comment:
I don't know (it can be removed, it seems). I copied it from the myst3 engine, which is also declared as static:

https://github.com/scummvm/scummvm/blob/5a5446dc64920055141724c8c8751723ea8578c9/engines/myst3/gfx.h#L93-L105

Perhaps @aquadran knows?

Reply:
I think this was declared as static to allow the renderer to query the pixel format without needing to create an actual texture.

Reply:
yes, that is purpose for that

Reply:
Then it should stay, since it is used to know which pixel format to use, depending on the renderer.